include/simdjson/common_defs.h

```diff
@@ -77,9 +77,12 @@ constexpr size_t DEFAULT_MAX_DEPTH = 1024;
   #define unlikely(x) x
   #endif
 
+  #include <CppCoreCheck\Warnings.h>
   #define SIMDJSON_PUSH_DISABLE_WARNINGS __pragma(warning( push ))
   #define SIMDJSON_PUSH_DISABLE_ALL_WARNINGS __pragma(warning( push, 0 ))
   #define SIMDJSON_DISABLE_VS_WARNING(WARNING_NUMBER) __pragma(warning( disable : WARNING_NUMBER ))
+  // Get rid of Intellisense-only warnings (Code Analysis)
+  #define SIMDJSON_DISABLE_UNDESIRED_WARNINGS SIMDJSON_DISABLE_VS_WARNING(ALL_CPPCORECHECK_WARNINGS)
```

Comment:
I expect that you did review the changes (there were 3 warning types?) and none of the jumped at you as fault in our code, right?

Reply:
Yeah. Of the three types:

```
Error (active)	E0960	type used as constructor name does not match type "simdjson::haswell::simd::simd8<bool>"
```

I couldn't make heads or tails of this--the names match character for character, including template arguments, and the Internet was unhelpeful--and decided it was a purely syntactic waste of time.

```
Warning	C26451	Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow (io.2).
```

This was the most potentially real, but everywhere I found it, it was super deliberate and also fixing it to use casts seemed like it might be in danger of regressing optimizations other compilers had done (I've seen things get really fiddly when you change where you do casting).

```
Warning	C26812	The enum type 'simdjson::error_code' is unscoped. Prefer 'enum class' over 'enum' (Enum.3).
```

Like, that's not wrong, except it's kind of our public API and I haven't felt comfortable changing it (kinda want to not export words like SUCCESS tho).